tools/cldr-apps/js/src/esm/cldrBulkConverter.mjs

```diff
@@ -0,0 +1,243 @@
+import * as XLSX from "xlsx";
+import * as cldrLoad from "./cldrLoad.mjs";
+
+/**
+ * Read an XLSX (or whatever) file.
+ * @param {File} file file object
+ * @param {Function} cb callback for returning data: (err, wb)
+ * @returns Promise
+ */
+function xlsUpload(file, cb) {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      try {
+        // todo: err?
+        const data = new Uint8Array(reader.result);
+        const wb = XLSX.read(data, { type: "array" });
+        cb(null, wb);
+        return resolve;
+      } catch (err) {
+        cb(err);
+        return reject(err);
+      }
+    };
+    reader.readAsArrayBuffer(file);
+  });
+}
+
+/**
+ * read the header row from the sheet
+ * @param {XLSX.WorkBook} wb
+ */
+function xlsHeaders(wb) {
+  if (!wb) throw Error(`No workbook`);
+  const sheet = wb.Sheets[wb.SheetNames[0]];
+  const headers = [];
+
+  for (let c = 0; ; c++) {
+    const cell_ref = XLSX.utils.encode_cell({ c, r: 0 });
+    if (!sheet[cell_ref]?.v) {
+      break;
+    }
+    headers.push(sheet[cell_ref].v);
+  }
+  return headers;
+}
+
+function xlsGenerateXml(locale, wb, xColumn, vColumn) {
+    if (!wb) throw Error(`No workbook`);
+    const sheet = wb.Sheets[wb.SheetNames[0]];
+    const headers = xlsHeaders(wb);
+  const xNumber = headers.findIndex((v) => v === xColumn); // col # for XPath
+  const vNumber = headers.findIndex((v) => v === vColumn); // col # for Value
+
+  if (xNumber === -1 || vNumber === -1 || xNumber === vNumber) {
+    throw Error(`xpath and value columns must be present and distinct`);
+  }
+
+  const locmap = cldrLoad.getTheLocaleMap();
+  const info = locmap.getLocaleInfo(locale);
+  const { bcp47, name } = info;
+  const { language, script, region, variant } = new Intl.Locale(bcp47);
+
+  // ok make it up.
+  // skeletal XML for ordering
+  const xml = `<?xml version="1.0" encoding="UTF-8" ?>
+<!-- Don't change the DOCTYPE. -->
+<!DOCTYPE ldml SYSTEM "../../common/dtd/ldml.dtd">
+<!-- Generated using cldrBulkConverter from a spreadsheet
+     locale: ${locale} / ${bcp47} / ${name}
+-->
+
+<!-- Copyright © 1991-2023 Unicode, Inc.
+For terms of use, see http://www.unicode.org/copyright.html
+SPDX-License-Identifier: Unicode-DFS-2016
+CLDR data files are interpreted according to the LDML specification (http://unicode.org/reports/tr35/)
+-->
+<ldml>
+	<identity>
+        <version number="$Revision$"/><!-- don't remove this.-->
+    </identity>
+
+    <characterLabels/>
+    <localeDisplayNames>
+        <languages/>
+		<scripts/>
+        <territories/>
+        <types/>
+    </localeDisplayNames>
+    <units>
+        <unitLength type="long"/>
+    </units>
+    <annotations/>
+    <characters/>
+    <numbers>
+        <miscPatterns numberSystem="latn"/>
+    </numbers>
```

Comment:
realized this skeleton needs to be here to keep the XML in order… almost 100% from client!

Reply:
fixed w/ leniency on cldrfile